packages/block-library/src/button/editor.scss

```diff
@@ -91,3 +75,21 @@
 		margin-top: $grid-size-large;
 	}
 }
+
+// Tweak styles for the Preview.
+.block-editor-block-preview__content {
```

Comment:
I don't think we should be having styles per block for previews. I think the preview should be generic enough. The problem with these styles is that the block can be nested in templates and these templates can be previewed too. I'm pretty sure this would break the previews of the templates using buttons.

Reply:
What I'm trying to do here is applying styles only when the blocks are being previewed. The styles defined below are needed to adjust the preview for the button block. Without them, the preview gets weird positions, especially when it sets the button alignment.

If I move these rules out of the `.block-editor-block-preview__content` scope, the styles will be applied to all places where the blocks are inserted, which is undesired.

> The problem with these styles is that the block can be nested in templates and these templates can be previewed too.

If I'm not wrong, these templates always will contain the `.block-editor-block-preview__content` css class into its content, so it means that these styles will be always applied, as long as the blocks are previewed using the `<BlockPreview />` block.

Reply:
But what I'm saying is that they probably shouldn't be applied when previewing templates. They should only be applied when previewing a single button block.

Also, I feel like this is not a generic enough fix. Imagine a small third party block, will we force them to have special styles for the preview?

Isn't there any way to fix the "position" of the scaled preview based on the scaled element? Right now we're trying to center horizontally, even aligning left would be better (it won't solve right aligned blocks though that's why I think the ideal solution is just to align the "left" of the preview with the "left" of the scaled element)

Reply:
> But what I'm saying is that they probably shouldn't be applied when previewing templates. They should only be applied when previewing a single button block.

👍

Reply:
> Isn't there any way to fix the "position" of the scaled preview based on the scaled element?

it seems to be kind tricky if we want to keep all the variants: button text (size), alignments, etc.
What do you think reducing the scope of these rules only into the styles preview?

Reply:
Not sur eI follow why it's tricky. Here's some pseudo code to understand my idea.

```js
const scaledElementRect = element.getBoundingClientRect(); // element is the scaled  element, the first child of the block
const rootOfPreviewRect = previewRoot.getBoundingClientRect(); // previewRoot is the root element of the BlockList

const diff = scaledElementRect.left - rootOfPreviewRect.left; 
// And then apply this diff as a CSS transform or left property of the preview
````